tests/ImportResolverTest.php

```diff
@@ -51,4 +51,45 @@ public function testAliasedImport()
         $this->assertResolve(Tools::class . '\A', 'Tools\A');
         $this->assertResolve(Tools::class . '\A\B', 'Tools\A\B');
     }
+
+    /**
+     * @expectedException        \InvalidArgumentException
+     * @expectedExceptionMessage Cannot infer the file name from the given ReflectionObject
+     */
+    public function testConstructorWithInvalidInferFileNameShouldThrowInvalidArgumentException()
+    {
+        $resolver = new ImportResolver(new \ReflectionObject(new \Exception));
+    }
+
+    public function testConstructorWithReflectionProperty()
+    {
+        $resolver = new ImportResolver(new \ReflectionProperty(ReflectionTarget::class, 'foo'));
+
+        $this->assertSame(ReflectionTarget::class, $resolver->resolve('ReflectionTarget'));
+    }
+
+    public function testConstructorWithReflectionMethod()
+    {
+        $resolver = new ImportResolver(new \ReflectionMethod(ReflectionTarget::class, 'publicStaticMethod'));
+
+        $this->assertSame('Brick\Reflection\Tests\publicStaticMethod', $resolver->resolve('publicStaticMethod'));
```

Comment:
This test is actually weird, you're attempting to resolve a class called `publicStaticMethod` which is very confusing. Let's just call it 'Foo'?

Reply:
I write the wrong assertions here and it should be ```ReflectionTarget``` class.

Reply:
Ok, FYI it can be any class name, even if it does not exist!